lib/chat_bot.ex

```diff
@@ -126,7 +126,25 @@ defmodule Telegram.ChatBot do
               | {:ok, next_chat_state :: chat_state(), timeout :: timeout()}
               | {:stop, next_chat_state :: chat_state()}
 
-  @optional_callbacks handle_resume: 1, handle_info: 4, handle_timeout: 3
+  @doc """
+  Allows a chat bot to customize how updates are handled by this bot.
+
+  A return value of {:ok, %{"id" => id, ...}} will result in a new instance
+  of the bot being spun up to handle the update as a full chat.
+
+  A return value of {:transient, %{"id" => id, ...}} will result in a transient
+  instance of the bot being spun up to handle the update. In this case the process
+  will only exist to handle this specific update.
+
+  A return value of nil causes the update to be ignored.
+
+  This callback is optional and if not implemented the legacy behavior will
+  be used. The returned chat must have at least an entry containing the chat
+  id
+  """
+  @callback get_chat(update_type :: String.t(), update :: Types.update()) :: {:ok, Types.chat()} | {:transient, Types.chat()} | nil
+
+  @optional_callbacks get_chat: 2, handle_resume: 1, handle_info: 4, handle_timeout: 3
```

Comment:
We can directly add the `get_chat` default implementation down here in the `defoverridable` functions